### ISSUE TITLE
`kubernetes_service(_v1)` resource is missing `spec.port.appProtocol`

### ISSUE THREAD

**Issue by kriswuollett** [hashicorp/terraform-provider-kubernetes#1554]
### Description

The `kubernetes_service(_v1)` resource is missing `spec.port.appProtocol` property as seen in [Service v1 kubernetes-api v1.23](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#serviceport-v1-core) which is needed for deploying [`ingress-nginx`](https://raw.githubusercontent.com/kubernetes/ingress-nginx/controller-v1.1.0/deploy/static/provider/cloud/deploy.yaml) using `kubernetes_service` instead of defining the `Service` through a `kubernetes_manifest` resource.

**Comment by arybolovlev** [comment]
Hi @kriswuollett,

Thank you for reporting this issue. I am working on it within this [PR](https://github.com/hashicorp/terraform-provider-kubernetes/pull/1668) and have it attached to this issue. Once PR is ready and reviewed, it will be merged into the main branch and this issue will be automatically closed.

Please keep an eye on the release notes.

Thanks!
